Makefile

```diff
@@ -0,0 +1,58 @@
+# Image URL to use all building/pushing image targets
+IMG ?= quay.io/observatorium/observatorium-operator:latest
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# Install CRDs into a cluster
+install: manifests
+	kubectl apply -f deploy/crds
+
+# Uninstall CRDs from a cluster
+uninstall: manifests
+	kubectl delete -f  deploy/crds
+
+# Generate manifests e.g. CRD, RBAC etc.
+manifests: controller-gen
+	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=deploy/crds
+
+# Run go fmt against code
+fmt:
+	go fmt ./...
+
+# Run go vet against code
+vet:
+	go vet ./...
+
+# Generate code
+generate: controller-gen
+	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths="./..."
+
+# Build the docker image
+docker-build:
+	docker build . -t ${IMG}
+
+# Push the docker image
+docker-push:
+	docker push ${IMG}
+
+# find or download controller-gen
+# download controller-gen if necessary
+controller-gen:
+ifeq (, $(shell which controller-gen))
+	@{ \
+	set -e ;\
+	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
+	cd $$CONTROLLER_GEN_TMP_DIR ;\
+	go mod init tmp ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.4 ;\
+	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
+	}
+CONTROLLER_GEN=$(GOBIN)/controller-gen
+else
+CONTROLLER_GEN=$(shell which controller-gen)
```

Comment:
this is probably not a great idea since the generators vary quite a bit between different versions. If we cannot be sure of the installed version then we cannot guarantee the build will work

Reply:
This was generated by kubebuilder. How would you like to handle this?

Reply:
I'd prefer to install this dependency the same way we install go dependencies in all of our other projects, e.g. up, observatorium, thanos-receive-controller:
https://github.com/observatorium/observatorium/blob/master/tools.go
https://github.com/observatorium/observatorium/blob/master/Makefile#L118

Reply:
Done